src/toil/test/cwl/cwlTest.py

```diff
@@ -807,14 +814,94 @@ def test_kubernetes_cwl_conformance_with_caching(self):
             ),
         )
 
+    @slow
+    @needs_wes_server
+    def test_wes_server_cwl_conformance(self):
+        """
+        Run the CWL conformance tests via WES. TOIL_WES_ENDPOINT must be
+        specified. If the WES server requires authentication, set TOIL_WES_USER
+        and TOIL_WES_PASSWORD.
+
+        To run manually:
+
+        TOIL_WES_ENDPOINT=http://localhost:8080 \
+        TOIL_WES_USER=test \
+        TOIL_WES_PASSWORD=password \
+        python -m pytest src/toil/test/cwl/cwlTest.py::CWLv12Test::test_wes_server_cwl_conformance -vv --log-level INFO --log-cli-level INFO
+        """
+        endpoint = os.environ.get("TOIL_WES_ENDPOINT")
+        extra_args = [f"--wes_endpoint={endpoint}"]
+
+        # These are the ones that currently fail:
+        #   - 310: mixed_version_v10_wf
+        #   - 311: mixed_version_v11_wf
+        #   - 312: mixed_version_v12_wf
+        #   - 331: iwd-fileobjs1
+        #   - 332: iwd-fileobjs2
+
+        # Main issues:
+        # 1. `cwltool --print-deps` doesn't seem to include secondary files from the default
+        #     e.g.: https://github.com/common-workflow-language/cwl-v1.2/blob/1.2.1_proposed/tests/mixed-versions/wf-v10.cwl#L4-L10
+        # 2. We need a way to reinterpret relative paths inside the workflow files
+        #    e.g.: https://github.com/common-workflow-language/cwl-v1.2/blob/1.2.1_proposed/tests/iwd/iwd-fileobjs1.cwl#L7-L11
```

Comment:
Let me know if you'd like assistance to either of these. I could fix `cwltool --print-deps` to include defaults, for example.

Reply:
I would really appreciate if you could fix `cwltool --print-deps` to include the defaults! I don't think I am familiar with `cwltool` enough to be able to fix that :(. I should be able to fix the second issue because it has something to do with the new WES runner.

Reply:
Okay, I'll work on that today!

This runner is very helpful, thank you! I had a hacky bash script that I discarded in favor of your code. With your WES runner I have successfully run all the conformance tests for CWL versions 1.0 and 1.1 against local and remote installations of toil 5.6.0.

Would it be okay with you if later I split this out as a separate `cwl-wes-runner` project inside one of the CWL GitHub organizations? I'd then remove the code from `workflow-service` to use it instead.

Reply:
fixing `cwltool --print-deps` to add the secondary files when there is a default `clas: File` literal is going to be a lot of work; it is a very obscure situation that can be worked around by adding the `secondaryFiles` manually to the `class: File` literal, so feel free to skip those tests for now. The standalone `cwl-wes-runner` is more important, so I'm going to focus on that first.